### PR TITLE
Parquet: Refactor decompression, including more complete datapage v2 support

### PR DESCRIPTION
**Pull request by wisp3rwind**
I went ahead on https://github.com/duckdb/duckdb/discussions/4437: This refactors parquet decompression to share code, with the additional benefit of supporting all compression algorithms on v2 data pages.
(The last commit is unrelated, but from my understanding this size should really be validated instead of trusting the snappy header. I could move that into a separate PR.)

The PR compiles, but I have not actually tested it up to now. Does this look like the right approach? Then, I would look into adding a few additional parquet files that test the newly supported combinations of compression algorithm/data page version.

This should be easiest to review commit-by-commit.

### CODE REVIEW

**Comment by wisp3rwind** [comment]
> Thanks for the PR! Could you include some tests for the new functionality?

Sure! I just wanted to get some quick feedback (no thorough review) before spending the time on writing tests.

**Review by Mytherin — COMMENTED** [review]
The code looks fine to me in principle but I would still really like to see tests added before merging.

File: extension/parquet/column_reader.cpp [hunk 1]
```diff
@@ -188,95 +188,93 @@ void ColumnReader::PrepareRead(parquet_filter_t &filter) {
...
+
+	DecompressInternal(chunk->meta_data.codec, (const char *)compressed_buffer.ptr, page_hdr.compressed_page_size,
+	                   (char *)block->ptr, page_hdr.uncompressed_page_size);
+}
+
+void ColumnReader::DecompressInternal(CompressionCodec::type codec, const char *src, idx_t src_size, char *dst,
+                                      idx_t dst_size) {
+	switch (codec) {
 	case CompressionCodec::UNCOMPRESSED:
+		D_ASSERT(false);
```
[the remark this answers is not in the document]
Comment (wisp3rwind) [reply extension/parquet/column_reader.cpp:242]: Done (not yet pushed).

**Comment by wisp3rwind** [comment]
> The code looks fine to me in principle but I would still really like to see tests added before merging.

Absolutely; I've just not gotten around to finishing the tests (I have a WIP script to generate test data).

**Comment by wisp3rwind** [comment]
Is it possible to only run a subset of the sqllogictests when testing locally? https://duckdb.org/dev/testing does not mention anything.

**Comment by Mytherin** [comment]
> For example, if we wanted to execute only the above test, we would run the command unittest test/sql/projection/test_simple_projection.test. If we wanted to run all tests in a specific directory, we would run the command unittest "[projection]".

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**